esphome/components/qmc5883l/qmc5883l.cpp

```diff
@@ -60,10 +64,27 @@ void QMC5883LComponent::setup() {
     return;
   }
 
-  if (this->get_update_interval() < App.get_loop_interval()) {
+  uint32_t update_ms = this->get_update_ms_();
+  uint32_t loop_interval = App.get_loop_interval();
+  bool high_frequency = update_ms < loop_interval;
+  if (high_frequency)
```

Comment:
```suggestion
  uint32_t loop_interval = App.get_loop_interval();
  bool high_frequency = this->get_update_interval() < loop_interval || this->datarate_ >=  QMC5883L_DATARATE_100_HZ;
  if (high_frequency)
```
Maybe just remove get_update_ms_ and get_datarate_hz and do this?

Reply:
Updated suggestion

Reply:
Figured that out and fixed it except I didn't create an unnecessary variable.

Reply:
It wouldn't create an extra variable either way the compiler will optimize things. Don't have to worry about that.